yarn-project/end-to-end/src/simulators/token_simulator.ts

```diff
@@ -145,13 +157,15 @@ export class TokenSimulator {
       const wallet = this.lookupProvider.get(address.toString());
       const asset = wallet ? this.token.withWallet(wallet) : this.token;
 
-      const actualPrivateBalance = await asset.methods.balance_of_private({ address }).simulate();
+      const actualPrivateBalance = await asset.methods.balance_of_private({ address }).simulate({
+        skipTxValidation,
+      });
       expect(actualPrivateBalance).toEqual(this.balanceOfPrivate(address));
     }
   }
 
-  public async check() {
-    await this.checkPublic();
-    await this.checkPrivate();
+  public async check(skipTxValidation: boolean = false) {
```

Comment:
I'd make this be an options object with a single entry, or alternatively use enums. `check(true)` is not only quite cryptic, but it also conveys incorrect intuition - it sounds like you're actually checking the thing (because it's true!).